RestaurantsNearMe/Data/YelpRequest/RestaurantListProvider.swift

```diff
@@ -9,36 +9,61 @@ import Foundation
 import Networking
 import Combine
 
-struct RestaurantListProvider: RestaurantListProviding {
+class RestaurantListProvider: RestaurantListProviding {
   private let _restaurants = PassthroughSubject<[Restaurant], Never>()
   private let _errorMessage = PassthroughSubject<String, Never>()
+  private let _isDataLoading = PassthroughSubject<Bool, Never>()
 
   var restaurants: AnyPublisher<[Restaurant], Never> {
     _restaurants.eraseToAnyPublisher()
   }
   var fetchingError: AnyPublisher<String, Never> {
     _errorMessage.eraseToAnyPublisher()
   }
+  var isLoading: AnyPublisher<Bool, Never> {
+    _isDataLoading.eraseToAnyPublisher()
+  }
 
-  private let requestProvider: CurrentValueSubject<YelpRequest?, Never>
   private let client: AsyncHTTPClient
+  private let fetch = FetchingActor()
+  private var cancellables = Set<AnyCancellable>()
   
-  init(requestProvider: CurrentValueSubject<YelpRequest?, Never>, client: AsyncHTTPClient) {
-    self.requestProvider = requestProvider
+  init(requestProvider: AnyPublisher<YelpRequest?, Never>, client: AsyncHTTPClient) {
     self.client = client
+    requestProvider
+      .compactMap { $0 }
+      .sink { request in
+        Task { [weak self] in
+          await self?.updateRestaurants(for: request)
+        }
+      }.store(in: &cancellables)
+  }
```

Comment:
This is probably the biggest underlying switch. We essentially went from the client making *explicit* requests to the API to the underlying request being made when information changes. i.e viewModel doesn't need to call `networkingLayer.makeRequest()` but the view layer sends updates to combine which implicitly triggers a new API call.


This will likely evolve further as more things can trigger results like updating location.